EDL playbooks update

Ni-Knight opened the pull request:
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/46952

## Description
A playbook to search and tag CIDRs that are bigger that a given prefix and also tag TLDs wildcards. Tags are used to skip these IOCs when creating EDLs.

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

julieschwartz18 commented:
Doc review complete for the [CheckIfSubdomain.yml](https://github.com/demisto/content/pull/18523/commits/20886f371041f301ac7f14b209d15583ff5822d2) file. @Ni-Knight - please confirm the following:
- line 12 - the argument name is **left** - is this correct?
- line 15 - the argument name is **right** - is this correct?
- are the lists comma-separated? if so, that should be added to the description

julieschwartz18 commented:
Doc review complete for the Tag_massive_and_internal_IOCs_to_avoid_EDL_listing.yml file. @Ni-Knight - please confirm whether the following descriptions are ok:
- line 108 - the description for Add Big CIDRs to context is "Sets a value in context under the key you entered."
- line 235 - the description for Append Big_CIDR tag is "commands.local.cmd.add.values.to.indicator.multi.select.field" 
- line 23 - the description for Add Big DomainGlobs to context is "Sets a value in context under the key you entered."
- line 332 - the description for Append Big_Domainglob tag is "commands.local.cmd.add.values.to.indicator.multi.select.field"
- line 467 - the description for Checkmark Internal IPs is "commands.local.cmd.set.indicator"
- line 506 - the description for Add internal IPs to context is "Sets a value in context under the key you entered."
- line 593 - the description for Checkmark Internal Domains is "commands.local.cmd.set.indicator"
- line 632 - the description for Add internal domains to context is "Sets a value in context under the key you entered."
- line 690 - the description for Add internal CIDRs to context is "Sets a value in context under the key you entered."
- line 736 - the description for Add internal domains to context is "Sets a value in context under the key you entered."
- line 782 - the description for Append skip_edl tag tp IPs is "commands.local.cmd.add.values.to.indicator.multi.select.field"
- line 820 - the description for Append skip_edl tag to Domains is "commands.local.cmd.add.values.to.indicator.multi.select.field"

Ni-Knight commented:
> Doc review complete for the [CheckIfSubdomain.yml](https://github.com/demisto/content/pull/18523/commits/20886f371041f301ac7f14b209d15583ff5822d2) file. @Ni-Knight - please confirm the following:
> 
> * line 12 - the argument name is **left** - is this correct?
> * line 15 - the argument name is **right** - is this correct?
> * are the lists comma-separated? if so, that should be added to the description

Yes, its a filter so names are `left` and `right`.
List is not comma-separated:
![image](https://user-images.githubusercontent.com/24267931/163707713-1a32b04b-4c71-465a-a73d-1b58b06074c1.png)

Ni-Knight commented:
> should the second column in the first table be **Description** or maybe **Default Value**?

@julieschwartz18 - That is auto created by demisto-sdk so I would leave it like that.

Ni-Knight commented:
> Doc review complete for the Tag_massive_and_internal_IOCs_to_avoid_EDL_listing.yml file. @Ni-Knight - please confirm whether the following descriptions are ok:
> 
> * line 108 - the description for Add Big CIDRs to context is "Sets a value in context under the key you entered."
> * line 235 - the description for Append Big_CIDR tag is "commands.local.cmd.add.values.to.indicator.multi.select.field"
> * line 23 - the description for Add Big DomainGlobs to context is "Sets a value in context under the key you entered."
> * line 332 - the description for Append Big_Domainglob tag is "commands.local.cmd.add.values.to.indicator.multi.select.field"
> * line 467 - the description for Checkmark Internal IPs is "commands.local.cmd.set.indicator"
> * line 506 - the description for Add internal IPs to context is "Sets a value in context under the key you entered."
> * line 593 - the description for Checkmark Internal Domains is "commands.local.cmd.set.indicator"
> * line 632 - the description for Add internal domains to context is "Sets a value in context under the key you entered."
> * line 690 - the description for Add internal CIDRs to context is "Sets a value in context under the key you entered."
> * line 736 - the description for Add internal domains to context is "Sets a value in context under the key you entered."
> * line 782 - the description for Append skip_edl tag tp IPs is "commands.local.cmd.add.values.to.indicator.multi.select.field"
> * line 820 - the description for Append skip_edl tag to Domains is "commands.local.cmd.add.values.to.indicator.multi.select.field"

@julieschwartz18 - I've updated the description for the various tasks (those are our default descriptions which are fine but since it's new I updated them accordingly).

BEAdi commented on Packs/CommonScripts/Scripts/IsInCidrRanges/IsInCidrRanges.py:
```diff
@@ -1,19 +1,30 @@
+        bool: True if given IP is part of given CIDR range.
+    """
+
+    ip_addresses = argToList(demisto.args()['left'])
```
It is expected to be a list of IP addresses? If so, mention it in the yml and README files.

Ni-Knight replied:
Done

BEAdi commented on Packs/CommonScripts/Scripts/IsInCidrRanges/IsInCidrRanges.py:
```diff
@@ -1,19 +1,30 @@
+    ip_addresses = argToList(demisto.args()['left'])
     cidr_range_list = argToList(demisto.args()['right'])
 
-    for cidr in cidr_range_list:
-        if IPAddress(ip_address) in IPNetwork(cidr):
-            demisto.results(True)
-            return
+    for ip in ip_addresses:
+        for cidr in cidr_range_list:
+            if ipaddress.ip_address(ip) in ipaddress.ip_network(cidr):
+                demisto.results(True)
+                return
 
-    demisto.results(False)
+        demisto.results(False)
```
The result should be false only after we checked all IPs, isn't it?
So the previous indentation should be restored.

Ni-Knight replied:
I fixed the code, it gets two lists and checks if the values in list A are contained in at least one of the values in list B
![image](https://user-images.githubusercontent.com/24267931/165498785-3d6ddcd4-a499-4498-b847-eb49d45f6582.png)

Ni-Knight commented on Packs/CommonScripts/Scripts/CIDRBiggerThanPrefix/CIDRBiggerThanPrefix.py:
```diff
@@ -0,0 +1,34 @@
+
+def main():
+    ip_cidrs = argToList(demisto.args()['left'])
+    max_prefix = demisto.args()['right']
```
Moved the `int(max_prefix)` inside the try so if it fails it will be caught by exception as well.

Ni-Knight commented on Packs/CommonScripts/Scripts/CIDRBiggerThanPrefix/CIDRBiggerThanPrefix.yml:
```diff
@@ -0,0 +1,25 @@
+- name: right
+  required: true
+  description: The prefix to set as maximum.
+- name: left
+  required: true
+  description: A comma-separated list of CIDR to validate (IPv4 or IPv6).
```
Done

Ni-Knight commented on Packs/CommonScripts/Scripts/CIDRBiggerThanPrefix/CIDRBiggerThanPrefix.py:
```diff
@@ -0,0 +1,34 @@
+    return False
+
+
+def main():
```
Done

Ni-Knight commented on Packs/CommonScripts/Scripts/CheckIfSubdomain/CheckIfSubdomain.py:
```diff
@@ -0,0 +1,35 @@
+        Bool: True if domain is a subdomain of one of the internal domains
+    """
+
+    internal_domains = argToList(demisto.args()['right'])
+    domains = argToList(demisto.args()['left'])
```
The automation isn't taking the values from the list, it is getting it from the context so separator isn't `\n`

Ni-Knight commented on Packs/CommonScripts/Scripts/CheckIfSubdomain/CheckIfSubdomain.py:
```diff
@@ -0,0 +1,35 @@
+                demisto.results(True)
+                return
+
+        demisto.results(False)
```
It checks every domain in the list if its a subdomain of at least one of the domain listed as an asset. Answer is returned for each domain separately. I updated the code as it wasn't working as intended :(

BEAdi commented on Packs/CommonScripts/Scripts/CheckIfSubdomain/CheckIfSubdomain.py:
```diff
@@ -0,0 +1,39 @@
+from CommonServerPython import *  # noqa: F401
+
+
+def main():
```
Please also add here `try` and `except` that uses `return_error`, for edge cases.

Ni-Knight replied:
Done

BEAdi reviewed (changes requested):
Thank you for the changes you did, it looks great. I added some more pythonic changes to make the code more readable and easy to maintain.

Ni-Knight commented on Packs/CommonScripts/Scripts/IsInCidrRanges/IsInCidrRanges.yml:
```diff
@@ -7,19 +7,24 @@ type: python
 args:
 - name: left
   required: true
-  description: IPv4 address to filter.
+  description: The IPv4 address to search for.
 - name: right
   required: true
-  description: Comma-separated list of IPv4 ranges in CIDR notation against which to match.
+  description: A comma-separated list of IPv4 ranges in CIDR notation against which
+    to match.
```
Done

Ni-Knight commented on Packs/CommonScripts/Scripts/CheckIfSubdomain/CheckIfSubdomain.yml:
```diff
@@ -0,0 +1,25 @@
+  description: A list of domains to check if they are subdomains of the internal assets.
+- name: right
+  required: true
+  description: A list of internal assets (for example, mycompany.com)
```
Done

ShahafBenYakir commented on Packs/CommonScripts/ReleaseNotes/1_6_57.md:
```diff
@@ -0,0 +1,9 @@
+fetch_indicators_command
```
@Ni-Knight Is this on purpose?
